pallets/settlement/src/lib.rs

```diff
@@ -601,7 +601,7 @@ decl_module! {
             let did = Identity::<T>::ensure_perms(origin.clone())?;
             with_transaction(|| {
                 let portfolios_set = portfolios.into_iter().collect::<BTreeSet<_>>();
-                let legs_count = legs.iter().filter(|l| portfolios_set.contains(&l.from)).count() as u64;
+                let legs_count = legs.iter().filter(|l| portfolios_set.contains(&l.from)).count() as u32;
```

Comment:
I think then we also change the leg_id type, It should be `u32` as well.

Reply:
That requires a storage migration. I'll let it be like that for now.